SCHIZO/Helpers/StaticHelpers.cs

```diff
@@ -13,10 +14,10 @@ partial class CacheAttribute
     {
         public static void CacheAll()
         {
-            PLUGIN_ASSEMBLY.GetTypes().SelectMany(t => t.GetFields(BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic))
-                .Where(f => f.GetCustomAttribute<CacheAttribute>() != null)
-                .Select(f => (IDropdownList) f.GetValue(null))
-                .Select(d => d.Select(kvp => (string) kvp.Value))
+            PLUGIN_ASSEMBLY.GetTypes().SelectMany(t => t.GetMethods(BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic))
+                .Where(m => m.GetCustomAttribute<CacheAttribute>() != null)
+                .Select(m => (IEnumerable) m.Invoke(null, Array.Empty<object>()))
+                .Select(e => e.Cast<ITriDropdownItem>().Select(i => (string) i.Value))
                 .ForEach(CacheValues);
```

Comment:
what do you think about this
```suggestion
            IEnumerable<string> names = PLUGIN_ASSEMBLY.GetTypes()
                .SelectMany(t => t.GetMethods(BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic))
                .Where(m => m.GetCustomAttribute<CacheAttribute>() != null)
                .SelectMany(m => (IEnumerable<ITriDropdownItem>) m.Invoke(null, Array.Empty<object>()))
                .Select(i => i.Value.ToString());

            CacheValues(names);
```
or, if you want to enforce string dropdowns and error early instead of in CacheValues:
```suggestion
            IEnumerable<string> names = PLUGIN_ASSEMBLY.GetTypes()
                .SelectMany(t => t.GetMethods(BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic))
                .Where(m => m.GetCustomAttribute<CacheAttribute>() != null)
                .SelectMany(m => (IEnumerable<TriDropdownItem<string>>) m.Invoke(null, Array.Empty<object>()))
                .Select(i => i.Value);

            CacheValues(names);
```

Reply:
Wait you can cast do that kind of polymorphic cast directly? If the object extends `IEnumerable<TriDropdownItem<T>>` it directly extends `IEnumerable` but it only extends `IEnumerable<ITriDropdownItem>` indirectly. Does that cast work? (first suggestion)